### ISSUE TITLE
Travis CI broken - replace PPA with Snap installed LXD

### ISSUE THREAD

**Issue by cpu** [StreisandEffect/streisand#1150]
### Expected behavior:

CI runs nightly, creates a Streisand LXC and provisions it.

### Actual Behavior:

[CI fails all tasks](https://travis-ci.org/StreisandEffect/streisand/builds/331290341), failing to add the LXC PPA:

     TASK [Install LXC ppa] *********************************************************
     fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["add-apt-repository", "ppa:ubuntu-lxc/lxd-stable"], "delta": "0:00:00.957898", "end": "2018-01-20 20:38:01.876233", "failed": true, "msg": "non-zero return code", "rc": 1, "start": "2018-01-20 20:38:00.918335", "stderr": "", "stderr_lines": [], "stdout": "Cannot add PPA: 'ppa:ubuntu-lxc/lxd-stable'.\nPlease check that the PPA name or format is correct.", "stdout_lines": ["Cannot add PPA: 'ppa:ubuntu-lxc/lxd-stable'.", "Please check that the PPA name or format is correct."]}
	     to retry, use: --limit @/home/travis/build/StreisandEffect/streisand/tests/development-setup.retry

### Steps to Reproduce:

Invoke a build.

**Comment by cpu** [comment]
Made some progress on this today but after a few hours sunk into fixing the problem I'm still not 100% there.

Tl;dr - We missed a deprecation notice for the PPA: https://github.com/lxc/lxd/issues/4176 The official recommendation is to use the backported packages (too old for us in 14.04/travis) or install using `snap`.

@alimakki did some initial work to try and get a `snap` based LXD setup working in CI but ran into a unix socket error. I've made more progress since then and have LXD installed from snap working in CI and successfully booting a 16.04 container. The issue now is that the Ansible LXD connection to provision the container is failing. 

Excited to resume fighting this soon :sob:  but for today I'm out of time :ghost: :clock1:

**Comment by cpu** [comment]
Managed to solve the last outstanding issue last night. Fix is out in PR #1156
